### PR TITLE
close_corporation now closes corporations

### LINKED ISSUE #7527: [21MOON] Corporation closed from shares sold during emergency train buy remains open.
Game Id: 76944
https://18xx.games/game/76944

corporation closure occurs at: 
https://18xx.games/game/76944?action=389

What steps cause the unexpected behavior?

-7 Train is purchased and both LP and IPI corporations are trainless.
-LP takes its turn before IPI
-Corporation LP was forced to sell shares to purchase a train.
-Owner of LP sold 1 share of IPI, dropping its value from 40 to 0.
-Corporation IPI closes. its tokens are removed from the map and its money returned to treasury.
-IPI, though closed, is still showing on the turn order row and still holds onto owned treasury shares.
-On IPIs turn - instead of being skipped due to closure - it no longer runs, but its market value increases 2 steps [0 to 50] and then forces an emergency train buy.
-After the emergency train buy and the end of the turn, IPI's name finally disappears from the turn order listing, but it's token still remains on 50 spot in the market.

- Browser Version
  - On Chrome version 99.0.4844.57
- Platform: ChromeOS Desktop

Relevant game log section pulled from game:
[12:04] LP does not run
[12:04] LP's share price changes from ₡70 to ₡60
[12:04] iGuitar93 sells a 10% share of IPI and receives ₡40
**[12:04] IPI's share price changes from ₡40 to ₡0
[12:04] IPI closes**
[12:04] iGuitar93 sells a 10% share of SM and receives ₡50
[12:04] SM's share price changes from ₡50 to ₡40
[12:04] iGuitar93 contributes ₡185
[12:04] LP buys a 10 train for ₡730 from The Depot
[12:04] LP skips buy companies
[12:04] Zabur operates SM
[13:56] SM lays tile #20 with rotation 0 on K3
[13:59] SM lays tile #X17 with rotation 1 on I7
[14:01] SM places a token on D10 () for ₡25
[14:01] SM runs a 5 train for ₡290: L2-I7-E9-D10-B14 + E/W
[14:01] SM receives 'NE' end game bonus token
[14:01] SM receives 'SW' end game bonus token
[14:01] SM pays out ₡290 = ₡29 per share (₡145 to Zabur, ₡29 to indierokin)
[14:01] SM's share price changes from ₡40 to ₡60 (2 steps)
[14:01] SM skips buy trains
[14:01] SM skips buy companies
**[14:01] indierokin operates IPI**
[14:12] IPI passes lay/upgrade track
[14:12] IPI skips place a token
[14:12] IPI skips run routes
[14:12] IPI does not run
**[14:12] IPI's share price changes from ₡0 to ₡50 (2 steps)**
[14:15] IPI sells a 10% share of DSE and receives ₡140
[14:18] IPI sells a 10% share of SM and receives ₡60
[14:18] SM's share price changes from ₡60 to ₡50
[14:19] indierokin sells a 10% share of SM and receives ₡50
[14:22] indierokin sells a 10% share of DSE and receives ₡140
[14:22] indierokin contributes ₡530
[14:22] IPI buys a 10 train for ₡730 from The Depot
[14:22] IPI skips buy companies
[14:22] -- Operating Round 4.2 (of 2) --

Please include screenshots (you can drag and drop or upload them into this issue)
closed_IPI_still_in_turn_order

reopened_IPI_vanishes_at_end_of_OR

market_closed_to_50

### PR DESCRIPTION
**Pull request by roseundy**
Fixes #7527 

There is a chance this will invalidate a few games.

### CODE REVIEW

File: lib/engine/game/g_21_moon/game.rb [hunk 1]
```diff
@@ -920,6 +920,12 @@ def sellable_bundles(player, corporation)
           @round.active_step.sellable_bundles(player, corporation)
         end
 
+        def close_corporation(corporation, quiet: false)
```
Comment (tobymao) [inline lib/engine/game/g_21_moon/game.rb:923]: why don't you just fix the base class?

Comment (roseundy) [reply lib/engine/game/g_21_moon/game.rb:923]: I don't know what that will break. I assume there was a reason it didn't have it. I'll investigate.

**Comment by roseundy** [comment]
OK, I've changed the base close_corporation method to add a corporation.close! call
I've removed the now redundant corporation.close! calls from several games and removed now redundant close_corporation overrides from a few games.

File: lib/engine/game/g_21_moon/game.rb (hunk 1 above)
Comment (roseundy) [reply lib/engine/game/g_21_moon/game.rb]: See new commit.

**Pull request merged**